src/main.zig

```diff
@@ -171,7 +171,9 @@ fn linenoiseEdit(ln: *Linenoise, in: File, out: File, prompt: []const u8) !?[]co
 /// Read a line with custom line editing mechanics. This includes hints,
 /// completions and history
 fn linenoiseRaw(ln: *Linenoise, in: File, out: File, prompt: []const u8) !?[]const u8 {
-    defer out.writeAll("\n") catch {};
+    if (ln.print_newline) {
+        defer out.writeAll("\n") catch {};
+    }
```

Comment:
The `defer` has to be in the function scope, as `defers` work on a scope-basis, not function basis (e.g., like in go)